go/vt/wrangler/traffic_switcher.go

```diff
@@ -963,7 +1000,9 @@ func (ts *trafficSwitcher) checkJournals(ctx context.Context) (journalsExist boo
 func (ts *trafficSwitcher) stopSourceWrites(ctx context.Context) error {
 	var err error
 	if ts.MigrationType() == binlogdatapb.MigrationType_TABLES {
-		err = ts.changeTableSourceWrites(ctx, disallowWrites)
+		if err = ts.changeTableSourceWrites(ctx, disallowWrites); err == nil {
+			err = ts.manageTableLocksSourceWrites(ctx, tableLockAcquire)
```

Comment:
I feel like there has to be a `defer ts.manageTableLocksSourceWrites(ctx, tableLockRelease)`; since here is probably not the right function to have this `defer`, maybe the calling function should do it. Decoupling the two altogether makes me uneasy, because the code could break and we would not release the lock.

Reply:
Good call. I just added/improved error handling and added a deferred func for this there. Thank you!